src/graphics/program-lib/chunks/cubeMapProjectBox.frag

```diff
@@ -13,5 +15,6 @@ vec3 cubeMapProject(vec3 nrdir) {
 
     vec3 posonbox = vPositionW + nrdir * fa;
     vec3 envBoxPos = (envBoxMin + envBoxMax) * 0.5;
-    return posonbox - envBoxPos;
+
+    return (posonbox - envBoxPos) * vec3(-1, 1, 1);
```

Comment:
does this mean the dynamic cubemap wont be usable for box projection as that flips x?

Reply:
Box projection uses `cubeMapRotatePosX()` so doesn't flip with the rotate - but needs to flip here because we used to flip after `cubeMapProject()` - the end result is that for box projection, with no rotation the cost is still the same as before, but with rotation the cost isn't absorbed by the rotation.

Reply:
I've refactored the code - separating out rotation from -x flip